src/domain-engine.js

```diff
@@ -1,16 +1,16 @@
 import MetaDomain from './meta-domain'
 import getRegistration from './get-registration'
 import { get, set, createOrClone } from './utils'
-import { castPath } from './key-path'
+import { castPath, getKeyString } from './key-path'
 
 class DomainEngine {
   /**
    * @param {Microcosm} repo
    */
   constructor(repo) {
-    this.repo = repo
-    this.domains = []
     this.registry = {}
+    this.repo = repo
+    this.domains = [[[], this.repo]]
```

Comment:
ah, so `domains` is an array where every item is an array with the first element being the keypath for the domain, and the second being the domain (or Microcosm now) that registers to actions